src/phpDocumentor/Compiler/Pass/NamespaceTreeBuilder.php

```diff
@@ -102,13 +103,15 @@ protected function addElementsOfTypeToNamespace(ProjectDescriptor $project, arra
                 $this->addToParentNamespace($project, $namespace);
             }
 
+            assert($namespace instanceof NamespaceDescriptor);
```

Comment:
$project->getIndexes() return `Collection<DescriptorAbstract>`. I didn't find a static analysis magic yet to explain to phpstan that `->get('namespaces', new Collection())` mean it will get a `Collection<NamespaceDescriptor>`. The assert is only there to explain that to Phpstan, but it also controls that there isn't anything we don't want in the Collection. Ok, I'll use the Assert library